x86_symbolic/src/Data/Macaw/X86/Symbolic.hs

```diff
@@ -472,6 +480,102 @@ instance GenArchInfo LLVMMemory M.X86_64 where
     , updateReg = x86UpdateReg
     }
 
+-- | Helper, not exported
+writeBytesBackwards ::
+  MM.HasPtrWidth w =>
+  C.IsSymBackend sym bak =>
+  (?memOpts :: MM.MemOptions) =>
+  MM.HasLLVMAnn sym =>
+  bak ->
+  MM.MemImpl sym ->
+  MM.LLVMPtr sym w ->
+  Seq.Seq Word8 ->
+  IO (MM.LLVMPtr sym w, MM.MemImpl sym)
+writeBytesBackwards bak mem ptr bytes = do
+  let sym = C.backendGetSym bak
+  let i8 = MM.bitvectorType (Bytes.toBytes (1 :: Int))
+  z <- WI.natLit sym 0
+  one <- WI.bvOne sym ?ptrWidth
+  let writeByte (p, m) byte = do
+        p' <- MM.ptrSub sym ?ptrWidth p one
+        bv <- WI.bvLit sym (C.knownNat @8) (BVS.mkBV C.knownNat (fromIntegral byte))
+        m' <- MM.storeRaw bak m p' i8 CLD.noAlignment (MM.LLVMValInt z bv)
+        pure (p', m')
+  Monad.foldM writeByte (ptr, mem) (Seq.reverse bytes)
+
+-- | Create an allocation for the stack for x86_64 with the SysV ABI.
+--
+-- On x86_64 with the SysV ABI, the stack grows \"downwards\" from high
+-- addresses to low. The end of the stack is initialized with the ELF auxiliary
+-- vector (not modelled here), and functions expect the following data to be
+-- available above their stack frame (i.e., just above the address in @rsp@):
+--
+-- * Their stack-spilled arguments
+-- * The return address
```

Comment:
Perhaps these should be listed in the opposite order, since the return address is in `0(%rsp)`, followed by the stack-spilled arguments in `8(%rsp)`, `16(%rsp)`, and so on.

Reply:
Will make the relation of order to addresses more explicit in Haddocks.